commit 9dd9b68c4c44f0d9102eb85ee2fa36a8b7f638e3
Author: Harshula Jayasuriya <harshula@sgi.com>
Date:   Tue Feb 12 16:13:25 2008 -0500

    In mountd, if get_exportlist() (utils/mountd/mountd.c) returns NULL it
    should not be considered a failure. It just means that there are no
    exports on the system.
    
    The practical problem with the current code is that a showmount -e
    results in a syslog message from mountd that looks like:
    
    rpc.mountd: export request from 10.250.100.2 failed.
    
    Reviewed-by: Greg Banks <gnb@sgi.com>
    Signed-off-by: Harshula Jayasuriya <harshula@sgi.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/mountd/mountd.c b/utils/mountd/mountd.c
index 4a50588..63d5ce1 100644
--- a/utils/mountd/mountd.c
+++ b/utils/mountd/mountd.c
@@ -203,9 +203,8 @@ mount_dump_1_svc(struct svc_req *rqstp, void *argp, mountlist *res)
 	struct sockaddr_in *addr =
 		(struct sockaddr_in *) svc_getcaller(rqstp->rq_xprt);
 
-	if ((*res = mountlist_list()) == NULL)
-		xlog(L_WARNING, "dump request from %s failed.",
-			inet_ntoa(addr->sin_addr));
+	xlog(D_CALL, "dump request from %s.", inet_ntoa(addr->sin_addr));
+	*res = mountlist_list();
 
 	return 1;
 }
@@ -254,9 +253,8 @@ mount_export_1_svc(struct svc_req *rqstp, void *argp, exports *resp)
 	struct sockaddr_in *addr =
 		(struct sockaddr_in *) svc_getcaller(rqstp->rq_xprt);
 
-	if ((*resp = get_exportlist()) == NULL)
-		xlog(L_WARNING, "export request from %s failed.",
-			inet_ntoa(addr->sin_addr));
+	xlog(D_CALL, "export request from %s.", inet_ntoa(addr->sin_addr));
+	*resp = get_exportlist();
 		
 	return 1;
 }
@@ -267,9 +265,9 @@ mount_exportall_1_svc(struct svc_req *rqstp, void *argp, exports *resp)
 	struct sockaddr_in *addr =
 		(struct sockaddr_in *) svc_getcaller(rqstp->rq_xprt);
 
-	if ((*resp = get_exportlist()) == NULL)
-		xlog(L_WARNING, "exportall request from %s failed.",
-			inet_ntoa(addr->sin_addr));
+	xlog(D_CALL, "exportall request from %s.", inet_ntoa(addr->sin_addr));
+	*resp = get_exportlist();
+
 	return 1;
 }
 
